service-base/src/test/java/org/eclipse/hono/service/management/device/AutoProvisioningEnabledDeviceBackendTest.java

```diff
@@ -119,17 +117,12 @@ public void testProvisionDeviceRemovesDeviceIfCredentialsCreationFails(final Ver
         final AutoProvisioningEnabledDeviceBackend underTest = mock(AutoProvisioningEnabledDeviceBackend.class);
         when(underTest.provisionDevice(anyString(), any(), any())).thenCallRealMethod();
 
-        doAnswer(invocation -> {
-            final Promise<OperationResult<Id>> promise = invocation.getArgument(4);
-            promise.complete(OperationResult.ok(201, Id.of(DEVICE_ID), Optional.empty(), Optional.empty()));
-            return null;
-        }).when(underTest).createDevice(any(), any(), any(), any(), any(Handler.class));
+        when(underTest.createDevice(any(), any(), any(), any()))
+                .thenReturn(Future.succeededFuture(
+                        OperationResult.ok(201, Id.of(DEVICE_ID), Optional.empty(), Optional.empty())));
 
-        doAnswer(invocation -> {
-            final Promise<Result<Void>> promise = invocation.getArgument(4);
-            promise.complete(Result.from(204));
-            return null;
-        }).when(underTest).deleteDevice(any(), any(), any(), any(), any(Handler.class));
+        doAnswer(invocation -> Future.succeededFuture(Result.from(204)))
+                .when(underTest).deleteDevice(any(), any(), any(), any());
```

Comment:
Can be simplified (`when(underTest.deleteDevice)...`).

Reply:
@calohmn I have updated it. Also made changes regarding indentation as in the above comments. Would you mind taking a look again?